crates/sui-types/src/messages.rs

```diff
@@ -496,98 +497,59 @@ impl TransactionData {
-#[derive(Debug, Clone, Serialize, Deserialize)]
-#[serde(remote = "TransactionEnvelope")]
-pub struct TransactionEnvelope<S> {
-    // This is a cache of an otherwise expensive to compute value.
-    // DO NOT serialize or deserialize from the network or disk.
-    #[serde(skip)]
-    transaction_digest: OnceCell<TransactionDigest>,
-    // Deserialization sets this to "false"
-    // TODO: is_verified is only set to true in some callsites after verification.
-    // Hence it's not optimal.
-    #[serde(skip)]
-    pub is_verified: bool,
-
-    // The packet of data that authorities will sign on. Stores the tx data and the sender signature.
-    pub signed_data: SenderSignedData,
-
-    /// authority signature information, if available, is signed by an authority, applied on `tx_signature` || `data`.
-    pub auth_sign_info: S,
-    // Note: If any new field is added here, make sure the Hash and PartialEq
-    // implementation are adjusted to include that new field (unless the new field
-    // does not participate in the hash and comparison).
-}
-
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
 pub struct SenderSignedData {
     pub data: TransactionData,
     /// tx_signature is signed by the transaction sender, applied on `data`.
     pub tx_signature: Signature,
 }
 
-impl<S> TransactionEnvelope<S> {
-    #[allow(dead_code)]
-    fn add_sender_sig_to_verification_obligation(
-        &self,
-        obligation: &mut VerificationObligation,
-        idx: usize,
-    ) -> SuiResult<()> {
-        // We use this flag to see if someone has checked this before
-        // and therefore we can skip the check. Note that the flag has
-        // to be set to true manually, and is not set by calling this
-        // "check" function.
-        if self.is_verified || self.signed_data.data.kind.is_system_tx() {
-            return Ok(());
-        }
+impl Message for SenderSignedData {
+    type DigestType = TransactionDigest;
 
-        self.signed_data
-            .tx_signature
-            .add_to_verification_obligation_or_verify(self.signed_data.data.sender, obligation, idx)
+    fn digest(&self) -> Self::DigestType {
+        TransactionDigest::new(sha3_hash(self))
     }
 
-    pub fn verify_sender_signature(&self) -> SuiResult<()> {
-        if self.is_verified || self.signed_data.data.kind.is_system_tx() {
+    fn verify(&self) -> SuiResult {
+        if self.data.kind.is_system_tx() {
             return Ok(());
         }
-        self.signed_data
-            .tx_signature
-            .verify(&self.signed_data.data, self.signed_data.data.sender)
+        self.tx_signature.verify(&self.data, self.data.sender)
     }
 
+    fn add_to_verification_obligation(
+        &self,
+        _: &mut VerificationObligation,
+    ) -> SuiResult<()> {
+        panic!("CRITICAL: SenderSignedData::add_to_verification_obligation is called. This function is yet to be implemented and should never be called.")
```

Comment:
@lxfind I added this instead - this function should never be called. Is this alright?

Reply:
Also @lxfind ready for review.